src/pages/ProductDetail/components/DetailOptional/DetailOptional.js

```diff
@@ -0,0 +1,164 @@
+import { React, useState } from "react";
+import Calendar from "react-calendar/dist/umd/Calendar";
+import "react-calendar/dist/Calendar.css";
+import styled from "styled-components";
+
+const DetailOptional = ({ itemInfo }) => {
+  const [value, onChange] = useState(new Date());
+  const [calendarOpen, setCalendarOpen] = useState(false);
+  const { rooms } = itemInfo;
+
+  const calendarHandle = () => {
+    setCalendarOpen(!calendarOpen);
+  };
+
+  return (
+    <SearchOptionContainer>
+      <ChoseOption>
+        <DateContainer>
+          <ChoseDate onClick={calendarHandle}>날짜 고르기</ChoseDate>
+          {calendarOpen ? (
+            <CalendarModal onChange={onChange} value={value} />
+          ) : null}
```

Comment:
삼항연산자 대신 논리곱 연산자를 쓰시면 될 것 같습니다.
```suggestion
          {isCalendarOpen && (
            <CalendarModal onChange={onChange} value={value} />
          )}
```